components/motor/gpio/motor_encoder.go

```diff
@@ -256,13 +257,41 @@ func (m *EncodedMotor) goForMath(ctx context.Context, rpm, revolutions float64)
 // If revolutions is 0, this will run the motor at rpm indefinitely
 // If revolutions != 0, this will block until the number of revolutions has been completed or another operation comes in.
 func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
-	m.opMgr.CancelRunning(ctx)
 	ctx, done := m.opMgr.New(ctx)
 	defer done()
 
 	goalPos, goalRPM, direction := m.goForMath(ctx, rpm, revolutions)
 
-	return m.goForInternal(ctx, goalRPM, goalPos, direction)
+	if err := m.goForInternal(ctx, goalRPM, goalPos, direction); err != nil {
+		return err
+	}
+
+	if revolutions == 0 {
+		return nil
```

Comment:
Put in a comment reminding the reader that calling `GoFor` with 0 revolutions will really go forever. This looks too much like "we've already gone 0 revolutions, so we've reached the target and should return immediately."